src/java.base/share/classes/java/lang/foreign/Linker.java

```diff
@@ -195,6 +196,18 @@
  *     <td style="text-align:center;">{@link MemorySegment}</td>
  * </tbody>
  * </table></blockquote>
+ * <p>
+ * Due to limited ABI specification coverage, all the native linker implementations limit the function
+ * descriptors that they support to those that contain only so-called <em>canonical</em> layouts. These layouts
+ * have the following restrictions:
```

Comment:
The third item in the list is for group size needing to be a multiple of the alignment constraint, the forth item refers to that, so I'll keep the ordered list.

WRT byte order: I think it's fine to just say "native byte order". Keep it simple. Or do you think there might be confusion about the meaning of "native"? (It looks like not even ByteOrder::nativeOrder explains that term: https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/nio/ByteOrder.html#nativeOrder())

"padding layout elements" is not quite right I think, since they also have to be the right size. I think saying just "padding" is more accurate (and simpler).

I'll switch to using plain links instead.

Reply:
Uploaded another version